linked_list/linked_list.py

```diff
@@ -9,71 +9,148 @@ def __init__(self, value, next_node = None):
 # Defines the singly linked list
 class LinkedList:
     def __init__(self):
-      self.head = None # keep the head private. Not accessible outside this class
+        self.head = None # keep the head private. Not accessible outside this class
 
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def get_first(self):
-        pass
+        if not self.head:
+            return None
+        
+        return self.head.value
 
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(self, value):
-        pass
+        new_node = Node(value)
+        new_node.next = self.head
+        self.head = new_node
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def search(self, value):
-        pass
+        if not self.get_first():
+            return False
+        
+        current_node = self.head
+
+        while current_node != None:

```

Comment:
Small code style note: generally prefer to use `is` when comparing to what we call singleton values like`True`, `False`, and `None` as sometimes the `==` operator can be redefined with the `__eq__` method and occasionally lead to unexpected behavior with these values.
```suggestion
        while current_node is not None:

```